### ISSUE TITLE
Deprecate or re-write

### ISSUE THREAD

**Issue by ghost** [backdrop-contrib/instagram_block#6]
According to the [Drupal project page](https://www.drupal.org/project/instagram_block), the Instagram API that this module uses has been deprecated. The Drupal module is no longer supported.

This module should be either deprecated, or re-written to use the new API: https://developers.facebook.com/docs/instagram-basic-display-api/

**Comment by stpaultim** [comment]
Makes sense. For the record, I'm using this module on one site and have not had problems so far, but that could change anytime.

According to: https://www.instagram.com/developer/ it may break for me in "early 2020."

My client is not updating their instagram account, so I may recommend disabling this block rather than updating it.

**Comment by ghost** [comment]
I needed a solution for a site I'm working on, so I wrote a new module after finding a really cool, simple jQuery library to use (instead of Instagram's complicated API): https://github.com/backdrop-contrib/instagram_feed

The reason I chose to write a new module instead of offering to re-write and maintain this one is because the new solution allows putting the Instagram feed into any element on the site (not specific just to blocks). So figured a new name would suit.

I recommend deprecating this module in favour of https://github.com/backdrop-contrib/instagram_feed.

**Comment by stpaultim** [comment]
That all makes sense to me. I am just curious about the process for deprecating a module.

The downside is the lack of upgrade path for existing users. However, I notice that there are only 3 active users according to BackdropCMS.org and one of them is probably my site. So, this might not be something to worry about too much. 

I like to talk about this on dev call. 

EDIT: Actually, usage seems to be hovering between 1 and 3 sites. Which either means that 1-2 sites using it are relatively inactive or there is one site using it (mine?) and occasional experimentation.

https://backdropcms.org/project/usage/instagram_block

**Comment by jenlampton** [comment]
When a module get's deprecated, we should state that at the top of the info file, and recommend an alternative module (when possible). There are some checkboxes on the node form on backdropcms.org that will enable us to mark a module as unsupported, but you'll need to coordinate with a backdropcms.org admin to have those updated (until we can get that bit automated)

**Comment by stpaultim** [comment]
I just tried to quickly test this module and am just getting a fatal error, with this in the log:

`Exception: Bad Request in InstagramRequest->request() (line 99 of /app/modules/instagram_block/instagram_block.lib.php).`

I may not have configured this correctly, but I suspect that this module is just broken and should be removed from BackdropCMS.org.

Is anyone using this module successfully anymore. Apparently, I used to be, but I'm pretty sure that it's no longer working on the site it used to. 

What is the process for deprecating a module?
